### ISSUE TITLE
BitMessage doesn't connect to the network on Whonix 14. 

### ISSUE THREAD

**Issue by Cottonwoodhill** [Bitmessage/PyBitmessage#1343]
Hello BitMessage's Community,

I'm trying to use Bitmessage on Whonix 14 but it won't connect. It did on the previous version Whonix-13.0.01.4 (with a yellow light in my case)

Please find below some screenshots of my current configuration : 

![58735d4cabfc2258bb82c133a4b6f4fb945cd6a5_1_690x493](https://user-images.githubusercontent.com/37047745/45647410-172e4780-bac6-11e8-92fc-b8a4138ae27c.jpeg)
![9a3709e6e5d5dd9a8da5050ddd422d935711aa5e](https://user-images.githubusercontent.com/37047745/45647416-1a293800-bac6-11e8-8fc0-b5f0c3eccc71.jpeg)

I suppose without being an expert that there is a bug with the last version.  In fact, when I as example click on "About" on the tab window, Bitmessage doesn't show me the version neither the update and so on I am using... Furtherway, it doesn't save some of the configuration options I set after quitting the application. (Without killing it)

Does anybody was able to use Bitmessage on Whonix 14 and got a connection ? If so what should I change ? If not, should I install another version ?

Thank you for your help and best regards.

**Comment by g1itch** [comment]
At least you may click "Help -> Contact support" and past here any version information. Do you use the v0.6 tip or it's still 0.6.3.2? Are you sure you have tor running on port 9050?

**Comment by Cottonwoodhill** [comment]
Hello g1itch,

Sorry if the procedure is inadequate, I thought it would be easier to post the documentation here like some print screens. 

I am using ver 0.6.3.2 and yes ma TOR in running on port 9050.

One more thing I would point out is that I can't close Bitmessage without killing the process... 

Thank you for your support and if you want me to delete the issue no problem. 

Best regards.

**Comment by g1itch** [comment]
0.6.3.2 is outdated and contains known bugs. Please try the latest source: https://github.com/Bitmessage/PyBitmessage/archive/95e300d7cae5036d188ce9acc5bad6c91287dca1.tar.gz

**Comment by Cottonwoodhill** [comment]
Excuse me, but when extracted your file, the version is still 0.6.3.2 on my machine...

**Comment by dimyme** [comment]
cottonwoodhill is rite, the support BM does not state the commit hash.

the 0.6 tip  says it is   0632  !  i.e.  hash of  _"Fix pylint Instance of 'str' has no 'ids' member issue"_

**Comment by g1itch** [comment]
Commit is shown only if you run from the git repo. If you still cannot connect check debug.log for any exceptions - it may help to investigate the reason.

@dimyme any issues here should be about the latest code.

**Comment by dimyme** [comment]
**Commit is shown only if you run from the git repo.**

thanks... I wondered why it sometimes worked, sometimes not ...

**Comment by Cottonwoodhill** [comment]
Thank you guys for your interest in solving my case ! Much appreciated :)

As I'm not an expert could you please @g1itch tell me where the debug.log is situated ?

So what should I do now ? Keeep the version you proposed me to install ?

Thanks in advance for your reply & help.

**Comment by g1itch** [comment]
Yes, run the latest source, look into program directory or `~/.config/PyBitmessage/` for debug.log. You can also run PyBitmessage from terminal - you'll see any exceptions in stderr.

**Comment by Cottonwoodhill** [comment]
Hello guys,

I downloaded https://github.com/Bitmessage/PyBitmessage/archive/0.6.3.2.tar.gz and decompressed is in my /home/user/PyBitmessage folder and was able to connect to network.

So, everything is fine on my side, thank you again for your support and your great work.

Cheers :+1:

**Comment by dimyme** [comment]
don't forget to disable intel ME spy engine on your machine.

**Comment by Cottonwoodhill** [comment]
> don't forget to disable intel ME spy engine on your machine.

It's disabled, thank you for the advice :)

Best regards

**Comment by dimyme** [comment]
thanks glitch

**Comment by 0brand** [comment]
Sorry I'm a little late to the party. I may be wrong but I believe there may be some confusion on which version is non-functional in Whonix and Debian 9.

>@CottonwoodhillI: downloaded https://github.com/Bitmessage/PyBitmessage/archive/0.6.3.2.tar.gz and decompressed it to my /home/user/PyBitmessage folder and was able to connect to network on Whonix14.

I don't think Cottonwoodhill is using the latest source (latest commit 95e300d) which I believe is downloaded using `git clone https://github.com/Bitmessage/PyBitmessage $HOME/PyBitmessage` ?

Cottonwoodhill and myself have been trying to troubleshoot this using Debian (no Tor) without any luck. BTW thanks for all the help on forum @Cottonwoodhill !!

I'll post logs when I fireup bitmessage again :wink:

**Comment by PeterSurda** [comment]
@0brand you may be right, it's better to use the current v0.6 branch code instead of 0.6.3.2, as some network-related bugs were fixed.

In order to check if your installation is ok, run `python checkdeps.py`, which will evaluate the dependencies. I test all code on Debian9 so there shouldn't be any Debian-specific bugs.

I would just keep Bitmessage running, it will eventually connect, I think the bootstrap infrastructure is overloaded.

**Comment by g1itch** [comment]
@0brand if you don't use tor, remove SOCKS5 proxy setting and probably check UPnP.

**Comment by 0brand** [comment]
>In order to check if your installation is ok, run python checkdeps.py, which will evaluate the dependencies. I test all code on Debian9 so there shouldn't be any Debian-specific bugs.

Git cloned the latest v0.6 branch and `python checkdeps.py` indicated all dependencies where resolved. Firewall was disabled and I checked UPnP. However, after running for  a total of 6  hours  - no connection.  

Installed in  users directory.

`python setup.py install --user`

` ~/.local/bin/pybitmessage`

These are _all_ the log files.

```~/.local/bin/pybitmessage
Creating new config files in /home/user/.config/PyBitmessage/
2018-10-02 17:28:07,092 - WARNING - Using default logger configuration
2018-10-02 17:28:07,735 - WARNING - /home/user/.namecoin/namecoin.conf unreadable or missing, Namecoin support deactivated
2018-10-02 17:28:07,735 - WARNING - There was a problem testing for a Namecoin daemon. Hiding the Fetch Namecoin ID button
2018-10-02 17:28:07,739 - WARNING - No indicator plugin found
2018-10-02 17:28:07,763 - WARNING - No notification.message plugin found
2018-10-02 17:28:12,768 - WARNING - No notification.sound plugin found
```
Also installed in system directory as root and deleted `knownnodes.dat` with same results. Whats interesting is I ran 0.6.3.2  634a49c  and connected very quickly.

**Comment by g1itch** [comment]
@0brand, @Cottonwoodhill, please check latest code. It seems you suffered from #1335.

**Comment by 0brand** [comment]
@g1itch You are correct. Tested on Debian 9 clearnet and Whonix 14 and BitMessage connects in seconds with the latest code. Thanks for all your hard work! I think this issue can be closed :)

**Comment by Cottonwoodhill** [comment]
@g1itch I second 0brand's confirmation. Everything works fine. 
Thank you for updating this thread and for your great job !

**Comment by g1itch** [comment]
Seems to be fixed in ef54bca.
